tools/shared/src/main/scala/org/scalajs/core/tools/javascript/Printers.scala

```diff
@@ -212,48 +288,67 @@ object Printers {
 
         case New(ctor, args) =>
           def containsOnlySelectsFromAtom(tree: Tree): Boolean = tree match {
-            case DotSelect(qual, _)     => containsOnlySelectsFromAtom(qual)
+            case DotSelect(qual, _) => containsOnlySelectsFromAtom(qual)
             case BracketSelect(qual, _) => containsOnlySelectsFromAtom(qual)
-            case VarRef(_)              => true
-            case This()                 => true
-            case _                      => false // in particular, Apply
+            case VarRef(_) => true
+            case This() => true
+            case _ => false // in particular, Apply
+          }
+          if (containsOnlySelectsFromAtom(ctor)) {
+            print("new ")
+            print(ctor)
+          } else {
+            print("new (")
+            print(ctor)
+            print(')')
           }
-          if (containsOnlySelectsFromAtom(ctor))
-            print("new ", ctor)
-          else
-            print("new (", ctor, ")")
           printArgs(args)
 
         case DotSelect(qualifier, item) =>
-          print(qualifier, ".", item)
+          print(qualifier)
+          print(".")
+          print(item)
 
         case BracketSelect(qualifier, item) =>
-          print(qualifier, "[", item, "]")
+          print(qualifier)
+          print('[')
+          print(item)
+          print(']')
 
         case Apply(fun, args) =>
           print(fun)
           printArgs(args)
 
         case Spread(items) =>
-          print("...", items)
+          print("...")
+          print(items)
 
         case Delete(prop) =>
-          print("delete ", prop)
+          print("delete ")
+          print(prop)
 
         case UnaryOp(op, lhs) =>
           import ir.Trees.JSUnaryOp._
-          print("(", (op: @switch) match {
-            case + => "+"
-            case - => "-"
-            case ~ => "~"
-            case ! => "!"
-
-            case `typeof` => "typeof "
-          }, lhs, ")")
+          print('(')
+          if (op == `typeof`) {
+            print("typeof ")
+          } else {
+            print((op: @switch) match {
+                          case + => '+'
+                          case - => '-'
+                          case ~ => '~'
+                          case ! => '!'
+                        })
```

Comment:
Also, consider the following rewriting, which don't need the special (unlikely) `if`:
```scala
(op: @switch) match {
  case +        => print('+')
  case -        => print('-')
  case ~        => print('~')
  case !        => print('!')
  case `typeof` => print("typeof ")
}
```